pennylane/ops/meta.py

```diff
@@ -201,17 +201,33 @@ def circuit():
     num_params = 0
     grad_method = None
 
-    def __init__(self, tag=None, measurement=None):
+    def __init__(self, tag=None, measurement=None, use_device_shots=False, shots=None):
```

Comment:
Yeah it could definitely be simplified, although I am against allowing a string value in a dominantly-integer valued parameter (it's gonna be a minor inconvenience with type hints later on).

I would suggest using the value `-1` instead as a placeholder for inferred shots. (Kinda like how it's done for `np.array.shape`)

Reply:
> Also, I believe using the device's shots by default would be breaking, since the default measurement is qml.state() and inferring shots from the device would break that

I don't agree with this, mostly because users already expect a state return with `qml.Snapshot` regardless of finite shot or analytic execution.